Add support for games other than Minetest & minor deprecation cleanups

maplemedley opened the pull request:
The deprecated things that were removed have been disabled for newer versions of Luanti, meaning there's no change in behaviour.
Tested on Minetest and Mineclonia.

maplemedley committed:
Use xcompat for recipes

maplemedley committed:
Use xcompat for sounds

maplemedley committed:
Remove deprecated formspec variables

These currently resolve to `""`, so this doesn't cause any changes.
See https://github.com/luanti-org/minetest_game/blob/master/mods/default/legacy.lua

maplemedley committed:
Update deprecated const usage

maplemedley committed:
Update README dependencies & commonmark formatting

maplemedley committed:
Increment version in README

maplemedley committed:
Fix changelog order for 0.3

wsor4035 commented on crafts.lua:
```diff
@@ -20,92 +20,100 @@
 local button = material_fallback({
   'mesecons_button:button_off',
-  'default:stone',
+  'mcl_buttons:button_stone_off',
```
is this name still valid between voxelibre and mineclonia?

maplemedley replied:
Unsure, if it isn't the `xcompat.materials.stone` will kick in like stone did before.

wsor4035 replied:
fair enough

wsor4035 commented on crafts.lua:
```diff
@@ -20,92 +20,100 @@
-minetest.register_craft({
-	output = "digiterms:lcd_monitor 1",
+core.register_craft({
+	output = "digiterms:lcd_monitor",
```
why are we dropping the 1?

maplemedley replied:
None of the other recipes had it, and I had assumed it defaulted to 1 when omitted, does this change the result? Feels like an alright change to make the recipe calls consistent if it doesn't.

wsor4035 replied:
hmmm, documentation doesnt explicitly say

maplemedley replied:
Tested it out out on Mineclonia & Minetest and it outputs 1 item regardless of "1" ending the recipe string, so it should be alright

wsor4035 reviewed (commented):
few questions

wsor4035 reviewed (approved):
outside checking the one thing, lgtm

maplemedley commented:
I'm gonna have a look at the LCD recipe

maplemedley committed:
Fix empty column requirement

With this the LCD can have an empty column on either its left or right

maplemedley committed:
Make glass material consistent with digilines and between games

Digiline recipes, including for the vanilla LCD panel, all use full glass blocks, so it seems sensible to switch to using the same here.
Additionally, xcompat and some games don't have glass panes, so doing this also unifies the glass across games.
It also means the player will have one less unnecessary step to do and inventory slot occupied to craft LCD's.

maplemedley committed:
Fix Mineclonia copper ingot support

maplemedley committed:
Additional fallback for silicon

maplemedley commented:
Tested on Minetest and Mineclonia.
Worth noting that the recipe now calls for a glass block, instead of glass panes. This is due to glass panes not being in xcompat or present in all games, because its frees up an extra crafting step and inventory slot taken up, and because glass blocks are what's used in Digilines recipes (including its baked-in LCD).

SwissalpS commented:
I don't fully agree with the switch to glass block in the recipe. I do however understand why. However it would make more sense to enhance [xcompat] than to bend knee to the younger mod.

maplemedley commented:
Expanding xcompat wouldn't be a big deal, although Hades doesn't have any glass panes so the recipes wouldn't be unified.
To elaborate on why I believe unifying the recipe using glass blocks here is the best approach, besides the Hades thing:
 
In my experience, It's always difficult to remember whether a recipe calls for glass panes or glass blocks, and every time a recipe calls for glass panes, you have to take a whole extra step to clear out the crafting table, have at least six glass, and then you fill up your inventory with 16 of a new block that you have no use for and no existing inventory slot or chest slot for. 

From what I've seen when playing with friends, most people also don't seem to have a dedicated chest slot for the pane variants of glass. At best its in an unsorted junk chest somewhere.

It'd only serve as a mandatory extra step and layer of friction, between the player and crafting digiterms, without the recipe being more expensive or more interesting or engaging, just an annoying manual chore. It feels like an example of the inventory bloat that has gotten so infamous in the other big block game, and of recipe-chore bloat without any rewarding progression tied to it, which mod soups for said game have also gotten a reputation for. 

The players will also be used to using full glass blocks for Digilines recipes, including the original flat Digilines LCD. This also means that they're likely to already have a slot in their inventory/chest dedicated to full glass blocks when crafting things for Digilines, and glass blocks will be the most intuitive to new players.


Additionally the ContentDB page also never described any recipe, and instead has asked the player to use the craft guide, so old players won't get lost with a change like this.

There won't be a more convenient time to do this than now when expanding support to more new players.

SwissalpS commented:
a) there is no need for games to adapt their recipes and behaviours to one another. I don't see any problem with Hades using blocks and the other games that support it useing panes.

b) I don't understand why you want to reduce crafting steps. Especially one that makes so much sense physically. Why get rid of a step that makes building factories for them a bit more interesting?

c) imo the digilines touchscreen recipe is on the wrong side of the argument. How come does a touchscreen need 7 cubic meters of glass?! Plus the 1.5 cubic meters from making the LCD panel.

d) having left over panes made me become more creative and find a way to use them. If they build up in your storage use the trash, glass is cheap.

e) every server can choose to override any recipe at any time if they consider crafting too hard for their audience, but also don't want to hand out 'creative' and 'give' privs :D

maplemedley commented:
Thanks for the response, here's what I think
a) Absolutely, but when we have the opportinity its a good idea to reduce complexity 
b) We shouldn't intentionally have pointless steps in recipes to make them marginally more involved for the 1% of people who want to create a factory for every item a mod provides (although its fun), like these computer monitors, when it makes the recipe somewhat less approachable for the other 99%.
c) Since this mod is an expansion of Digilines the cubic-meter-of-glass logic is already established, its like how we no longer think its weird that 8 cubic meters of plank makes one cubic meter of chest. 
d) Thats a good point and something to consider, but personally I still think reducing complexity and friction is more valuable for this tech-player focused mod.
e) Defaults and usability matter

---

Basically my stance comes down to this: There won't be a better opportunity for this than now, as thousands more players gain compatibility with the mod. Unifying its glass material with what the Digilines players are used to using for crafting LCDs would help with the intuitiveness and friction of the recipe. The best reason I see not to do it is to not disturb the prior users.

SwissalpS commented:
Would be great if some other devs and players chimed in, at least to the recip part. :D

wsor4035 commented:
i say send it 🤷

wsor4035 commented:
no other comments, merging
